modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,366 @@
+    private static final Logger LOG = Logger.getLogger(IncrementalDataPurging.class);
+    private long purgeExecutionInterval = Expression.Time.minute(5).value();
+    private long sec = Expression.Time.sec(30).value();
+    private long mins = Expression.Time.hour(24).value();
+    private long hours = Expression.Time.day(30).value();
+    private long days = Expression.Time.year(5).value();
+    private long months = -1;
+    private long years = -1;
+    private boolean purgingEnabled = true;
+    private Map<TimePeriod.Duration, Long> retentionPeriods = new EnumMap<>(TimePeriod.Duration.class);
+    private StreamEventPool streamEventPool;
+    private Map<TimePeriod.Duration, Table> aggregationTables;
+    private SiddhiAppContext siddhiAppContext;
+    private Map<String, Table> tableMap;
+    private static ScheduledFuture scheduledFuture;
+    private static final String INTERNAL_AGG_TIMESTAMP_FIELD = "AGG_TIMESTAMP";
+
+
+    public void init(AggregationDefinition aggregationDefinition, StreamEventPool streamEventPool,
+                     Map<TimePeriod.Duration, Table> aggregationTables, SiddhiAppContext siddhiAppContext,
+                     Map<String, Table> tableMap) {
+        this.siddhiAppContext = siddhiAppContext;
+        List<Annotation> annotations = aggregationDefinition.getAnnotations();
+        this.tableMap = tableMap;
+        this.streamEventPool = streamEventPool;
+        this.aggregationTables = aggregationTables;
+        for (TimePeriod.Duration duration : aggregationTables.keySet()) {
+            switch (duration) {
+                case SECONDS:
+                    retentionPeriods.put(duration, sec);
+                    break;
+                case MINUTES:
+                    retentionPeriods.put(duration, mins);
+                    break;
+                case HOURS:
+                    retentionPeriods.put(duration, hours);
+                    break;
+                case DAYS:
+                    retentionPeriods.put(duration, days);
+                    break;
+                case MONTHS:
+                    retentionPeriods.put(duration, months);
+                    break;
+                case YEARS:
+                    retentionPeriods.put(duration, years);
+            }
+        }
+
+        Map<String, Annotation> annotationTypes = new HashMap<>();
+
+        for (Annotation annotation : annotations) {
+            annotationTypes.put(annotation.getName().toLowerCase(), annotation);
+        }
+        if (annotationTypes.keySet().contains(SiddhiConstants.NAMESPACE_PURGE)) {
+            if (Objects.nonNull(annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                    getElement(SiddhiConstants.ANNOTATION_ELEMENT_ENABLE))) {
+                String enable = annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                        getElement(SiddhiConstants.ANNOTATION_ELEMENT_ENABLE);
+                if (!(enable.equals("true") || enable.equals("false"))) {
```

Comment:
I think these values don't have to be case sensetive. Therefore, shall we use equlaseIgnoreCase

Reply:
fixed